### DISCUSSION TITLE
Actions new ship 🚢  - macOS 14 (Sonoma) and the new M1 macOS runner

### DISCUSSION THREAD

**Discussion by Akash1134** [community/community#102846] [Actions]
We're excited to announce the launch of a new M1 macOS runner and introducing the new M1 macOS runner available to open source!

### The all new macOS 14 (Sonoma) is now available 💯 

This runner is available for all plans, free in public repositories, and eligible to consume included free plan minutes in private repositories. The new runner executes Actions workflows with a 3 vCPU, 7 GB RAM, and 14 GB of storage VM, which provides the latest Mac hardware Actions has to offer. The new runner operates exclusively on macOS 14 and to use it, simply update the `runs-on` key in your YAML workflow file to `macos-14`.

### Introducing the new M1 macOS runner available to open source! 🚀 

We are also making the macOS 14 runner image available to GitHub hosted runners. Workflows executed on this image will run exclusively on the 3 vCPU M1 runner announced earlier today. To use the runner, simply update the `runs-on:` key in your YAML workflow file to `macos-14`, `macos-14-xlarge`, or `macos-14-large`.

**Note**: The macOS 12 runner image will remain `latest` until migration of the `latest` YAML workflow label to macOS 14 in Q2 FY24 (April – June 2024). While macOS 13 is now generally available under the macos-13 label, this image will not be migrated to `latest`. Following this announcement, **macOS 11 runner image will begin deprecation immediately with retirement expected to complete by June 2024.**

The full list of software available for all macOS runner images can be found [here](https://github.com/actions/runner-images/tree/main/images/macos). If there is software you require that is not installed on the image, please create an issue in the [runner-images repository](https://github.com/actions/runner-images/issues).

**Comment by hfhbd** [comment]
Thank you! My build time is cut in half now. I hope many missing (native) arm binaries will be finally published.

**Reply by Akash1134** [reply]
Thanks for sharing @hfhbd Our product team would love to hear about this. We'll share your feedback around ARM binaries with them. ❤️

**Reply by CharlieBrownny** [reply]
tengo problemas porque no esta funcionando en mi MacBook PRO tengo el MAC OS monterrey 12.7.4. y lo estoy corriendo desde la terminal me arroja el siguiente error
si me pueden ayudar mi whatsapp es 52 3331057308 Carlos Moreno

I have problems because it is not working on my MacBook PRO, I have MAC OS Monterrey 12.7.4. and I am running it from the terminal it gives me the following error
If you can help me, my WhatsApp is 52 3331057308 Carlos Moreno

charlie@MacBook-Pro-de-Carlos bot-whats % node app06.js
Sesion Detectada
Cargando...
/Users/charlie/Documentos/bot-whats/node_modules/whatsapp-web.js/src/webCache/LocalWebCache.js:34
        const version = indexHtml.match(/manifest-([\d\\.]+)\.json/)[1];
                                                                    ^

TypeError: Cannot read properties of null (reading '1')
    at LocalWebCache.persist (/Users/charlie/Documentos/bot-whats/node_modules/whatsapp-web.js/src/webCache/LocalWebCache.js:34:69)
    at /Users/charlie/Documentos/bot-whats/node_modules/whatsapp-web.js/src/Client.js:771:36
    at processTicksAndRejections (node:internal/process/task_queues:96:5)
charlie@MacBook-Pro-de-Carlos bot-whats %

**Comment by xenshehz** [comment]
Firstly, I'd really like to thank Github for making M1 macOS runners available! They are incredibly useful!

For some reason, [ObjectCapture API](https://developer.apple.com/augmented-reality/object-capture/) is not supported on `macos-14`

Here is the current error I receive:

```zsh
Object Capture is not available on this computer.
Error: Process completed with exit code 1.
```

According to Apple's documentation this API should be included in macOS >= 12.0 SDK. Checking the runner readme for M1 (https://github.com/actions/runner-images/blob/main/images/macos/macos-14-arm64-Readme.md), it looks like it should be supported?

Any help in getting Object Capture API to work on macOS runners would be wonderful! Also, if this is not the right place to post this - please free to move it.

**Comment by geriux** [comment]
This is awesome! 🎉 Similar to the above, I tried to use this [in this PR](https://github.com/WordPress/gutenberg/actions/runs/7915125888/job/21606292419?pr=59070) and I encountered:

```
lerna ERR! - /Users/runner/work/gutenberg/gutenberg/node_modules/lerna/dist/cli.js
lerna ERR!     at Module._resolveFilename (node:internal/modules/cjs/loader:1144:15)
lerna ERR!     at Module._load (node:internal/modules/cjs/loader:985:27)
lerna ERR!     at Module.require (node:internal/modules/cjs/loader:1235:19)
lerna ERR!     at require (node:internal/modules/helpers:176:18)
lerna ERR!     at Object.<anonymous> (/Users/runner/work/gutenberg/gutenberg/node_modules/nx/src/native/index.js:135:29)
lerna ERR!     at Module._compile (node:internal/modules/cjs/loader:1376:14)
lerna ERR!     at Module._extensions..js (node:internal/modules/cjs/loader:1435:10)
lerna ERR!     at Module.load (node:internal/modules/cjs/loader:1207:32)
lerna ERR!     at Module._load (node:internal/modules/cjs/loader:1023:12)
lerna ERR! lerna Cannot find module '@nx/nx-darwin-arm64'
lerna ERR! lerna Require stack:
```

While installing the npm dependencies.

**Reply by boozook** [reply]
Time-to-time I'm getting errors about missed some pre-installed executable binaries. Totally strange behaviour!
[This for example.](https://github.com/boozook/playdate/actions/runs/8510241075/job/23308574457?pr=240#step:11:22)

**Comment by vinayhegde1990** [comment]
@Akash1134 based on the [full list](https://github.com/actions/runner-images/tree/main/images/macos) mentioned in [your comment](https://github.com/orgs/community/discussions/102846#discussion-6160476) - could you please confirm if the `macos-13-arm64` tag can be used to have a runner with the M1 architecture and MacOS v13.x.x in GitHub Actions workflows on the **Free** Pricing Plan?

**Comment by allcreatures** [comment]
When attempting to use macos-14-large on a Free plan at GitHub, with a public repository, we are seeing this error:

> The job was not started because recent account payments have failed or your spending limit needs to be increased. Please check the 'Billing & plans' section in your settings.

Are Beta images not included in Free plans?

**Reply by hfhbd** [reply]
`macos-14` is free, `macos-14-large` using more CPUs is not.

**Reply by allcreatures** [reply]
Thank you @hfhbd. Of course now that you have answered, I clearly see the big banner that states "Larger runners are only available for organizations and enterprises using the GitHub Team or GitHub Enterprise Cloud plans."
